src/XR/features/WebXRDOMOverlay.ts

```diff
@@ -0,0 +1,139 @@
+import { Tools } from "../../Misc/tools";
+import { Nullable } from "../../types";
+import { WebXRFeatureName, WebXRFeaturesManager } from "../webXRFeaturesManager";
+import { WebXRSessionManager } from "../webXRSessionManager";
+import { WebXRAbstractFeature } from "./WebXRAbstractFeature";
+
+/**
+ * Options for DOM Overlay feature
+ */
+export interface IWebXRDOMOverlayOptions {
+    /**
+     * Fail the Web XR Session if DOM overlay isn't available (TODO: would like to mark optional as a feature parameter)
+     */
+    // required?: boolean;
+
+    /**
+     * DOM Element for overlay into WebXR scene.
+     *
+     * NOTE: UA may make this element background transparent in XR.
+     */
+    element: Element;
+    /**
+     * Supress XR Select events on container element (DOM blocks interaction to scene).
+     */
+    supressXRSelectEvents?: boolean
+}
+
+/**
+ * DOM Overlay Feature
+ * 
+ * @since 5.0.0
+ */
+export class WebXRDOMOverlay extends WebXRAbstractFeature {
+
+    private xrDOMOverlayType: XRDOMOverlayType | null;
```

Comment:
nit - private vars with `_`

```suggestion
    private _xrDOMOverlayType: Nullable<XRDOMOverlayType>;
```